modules/asm/main.tf

```diff
@@ -51,6 +55,16 @@ resource "kubernetes_config_map" "asm_options" {
   depends_on = [google_gke_hub_membership.membership, google_gke_hub_feature.mesh]
 }
 
+module "mesh_id" {
+  source  = "terraform-google-modules/gcloud/google"
+  version = "~> 3.1"
+
+  create_cmd_entrypoint  = "gcloud"
+  create_cmd_body        = "container clusters update ${var.cluster_name} --project=${var.project_id} --zone=${var.cluster_location} --update-labels=mesh_id=proj-${data.google_project.fleet.number}"
```

Comment:
This will cause a diff when users manage clusters via TF. If users are managing ASM via TF, it seems reasonable to assume they will be managing clusters via TF too which means they can follow the docs in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1284

Reply:
I see, so the suggestion is to keep this as a documented step instead of including in the module? I'm fine with that as well.

Reply:
Yes, I think docs is easiest. Something we could investigate is if we can fail fast and surface error using https://www.terraform.io/language/meta-arguments/lifecycle#custom-condition-checks. 

```tf
resource "kubernetes_namespace" "system" {
  metadata {
    name = "istio-system"
  }
    lifecycle {
    # The cluster must have mesh-id label
    precondition {
      condition     = contains( keys(data. google_container_cluster.asm.labels), "mesh_id") && maybe validate value too
      error_message = "The GKE cluster must have mesh_id label"
    }
  }
}
```
This will be a breaking change (due to min TF version bump) and require customers to upgrade to TF 1.2

Reply:
Sounds good. Do we know how long before we expect it's reasonable to allow 1.2 features in our modules? Looks like it just released <1 month ago.